### PR TITLE
feat/component_asset

### PR DESCRIPTION
**Pull request by boltsr**
-implement customized Asset
-Change with customized Asset

### CODE REVIEW

File: src/app/components/CurrentPositionProfit/index.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { toNumberFormat, weiToNumberFormat } from 'utils/display-text/format';
 import { useCurrentPositionPrice } from 'app/hooks/trading/useCurrentPositionPrice';
 import { LoadableValue } from '../LoadableValue';
 import { bignumber } from 'mathjs';
-
+import { AssetRenderer } from '../../components/CurrencyAsset';
```
Comment (creed-victor) [inline src/app/components/CurrentPositionProfit/index.tsx:14]: There should be new line between imports and declarations
```suggestion
import { AssetRenderer } from '../../components/CurrencyAsset';

```

File: src/app/components/Form/Input/index.tsx [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import React, { useCallback } from 'react';
 import cn from 'classnames';
 import { handleNumber } from 'utils/helpers';
-
+import { AssetRenderer } from '../../../components/CurrencyAsset';
+import { Asset } from 'types/asset';
 type InputType = 'text' | 'email' | 'password' | 'number';
```
Comment (creed-victor) [inline src/app/components/Form/Input/index.tsx:6]:
```suggestion
import { Asset } from 'types/asset';

type InputType = 'text' | 'email' | 'password' | 'number';
```

File: src/app/pages/BuySovPage/components/TopUpWallet/index.tsx [hunk 3]
```diff
@@ -7,7 +7,8 @@ import { Asset } from '../../../../../types/asset';
 import { FastBtcDialog } from '../../../../containers/FastBtcDialog';
 import { useTranslation } from 'react-i18next';
 import { translations } from '../../../../../locales/i18n';
-
+import { Trans } from 'react-i18next';
+import { AssetRenderer } from '../../../../components/CurrencyAsset';
 export function TopUpWallet() {
```
Comment (creed-victor) [inline src/app/pages/BuySovPage/components/TopUpWallet/index.tsx:12]:
```suggestion
import { AssetRenderer } from '../../../../components/CurrencyAsset';

export function TopUpWallet() {
```

**Review by creed-victor — CHANGES_REQUESTED** [review]

**Comment by boltsr** [comment]
got it

File: src/app/components/Form/Input/index.tsx [hunk 4]
```diff
@@ -38,6 +40,17 @@ export function Input({
...
+    if (appendElem === 'RBTC' || appendElem === 'rBTC') {
+      console.log('appendElem', appendElem);
+      return <AssetRenderer asset={Asset.RBTC} />;
+    } else if (appendElem === 'USDT') {
+      return <AssetRenderer asset={Asset.USDT} />;
+    } else {
+      return appendElem;
+    }
+  };
+
```
Comment (creed-victor) [inline src/app/components/Form/Input/index.tsx:53]: This file shouldn't have this change, instead you need to use component like this:

```
<Input {some_props} appendElem={<AssetRenderer asset={Asset.USDT} />} />
```

**Review by creed-victor — CHANGES_REQUESTED** [review]

**Review by creed-victor — APPROVED** [review]

**Pull request merged**